c2/cuda/groth16_cuda.cu

```diff
@@ -6,6 +6,10 @@
 #include <mutex>
 #include <algorithm>
 
+#define SPPARK_DONT_INSTANTIATE_TEMPLATES
```

Comment:
It would be appropriate to move this next to `#include <msm/pippenger.cuh>`. To emphasize the point where it's actually significant.

Same can be said about `typedef uint8_t byte;`, it's better moved to groth16_srs.cuh, next to the declaration where it's required.